framework/CodeInterfaces/PHISICS/phisicsdata.py

```diff
@@ -0,0 +1,961 @@
+        depLabelList, depList, timeStepList, matList = self.getDepInfo(timeStepIndex, mrtauTimeSteps, phisicsDataDict['numberOfMPI'])
+        decayLabelList, decayList = self.getDecayHeat(timeStepIndex,mrtauTimeSteps)
+        instantDict['keff'] = keff
+        instantDict['errorKeff'] = errorKeff
+        instantDict['reactionRateInfo'] = reactionRateInfo
+        instantDict['fissionMatrixInfo'] = fissionMatrixInfo
+        instantDict['workingDir'] = self.workingDir
+        instantDict['fluxLabelList'] = fluxLabelList
+        instantDict['fluxList'] = fluxList
+        instantDict['depLabelList'] = depLabelList
+        instantDict['depList'] = depList
+        instantDict['timeStepList'] = timeStepList
+        instantDict['decayLabelList'] = decayLabelList
+        instantDict['decayList'] = decayList
+        instantDict['timeStepIndex'] = timeStepIndex
+        instantDict['mrtauTimeSteps'] = mrtauTimeSteps
+        instantDict['xsLabelList'] = xsLabelList
+        instantDict['xsList'] = xsList
+        instantDict['cpuTime'] = cpuTime
+        if self.phisicsRelap is True:
+          instantDict['powerDensList'] = powerDensList
+          instantDict['powerDensLabelList'] = powerDensLabelList
+        if self.phisicsRelap is False:
+          instantDict['matFluxLabelList'] = matFluxLabelList
+          instantDict['matFluxList'] = matFluxList
+        self.writeCSV(instantDict,timeStepIndex,mrtauTimeSteps,phisicsDataDict['jobTitle'])
+
+      if phisicsDataDict['mrtauStandAlone'] == True:
+        decayHeatMrtau = self.getDecayHeatMrtau(timeStepIndex, mrtauTimeSteps)
+        depList = self.getDepInfoMrtau(timeStepIndex, mrtauTimeSteps)
+        mrtauDict['workingDir'] = self.workingDir
+        mrtauDict['decayHeatMrtau'] = decayHeatMrtau
+        mrtauDict['depList'] = depList
+        mrtauDict['timeStepIndex'] = timeStepIndex
+        mrtauDict['mrtauTimeSteps'] = mrtauTimeSteps
+        self.writeMrtauCSV(mrtauDict)
+  
+  def summedListValues(self,list):
+    """
+      Sums up values from a given list.
+      @ In, list, list of intergers or floats
+      @ Out, sum, integer or float, sum of the values given in input
+    """
+    sum = 0
+    for i in xrange (0,len(list)):
+      sum = sum + float(list[i])
+    return sum
+  
+  def summedDictValues(self,dict,typeOfParameter,parameterList):
+    """
+      Sums up the values from a given dictionary.
+      @ In, dict, dictionary, nested dictionaries of intergers or floats
+      @ In, typeOfParameter, string, either 'reactionRates' or 'FissionMatrix'
+      @ In, parameterList, list, list of the parameters parsed
+      @ Out, summedDict, dict, dictionary of integer or float
+    """
+    summedDict = self.declareDict(typeOfParameter,parameterList)
+    if self.printSpatialRR is False:
+      for parameter,mpi in dict.iteritems():
+        sum = 0
+        for mpi,value in mpi.iteritems():
+          sum = '%.15E' % Decimal(float(sum) + float(value))
```

Comment:
I do not get this...
Firstly you created a method for this
Secondly, you can remove the loop and have the following:
```
sumString =  '%.15E' % Decimal(  sum([float(elm) for elm in mpi.values()]) )

```

Reply:
whaty do you mean by: "Firstly you created a method for this"? Is there a niftier way to sum values in nested dictionaries?